lib/pure/terminal.nim

```diff
@@ -727,6 +727,13 @@ proc getch*(): char =
       doAssert(readConsoleInput(fd, addr(keyEvent), 1, addr(numRead)) != 0)
       if numRead == 0 or keyEvent.eventType != 1 or keyEvent.bKeyDown == 0:
         continue
+      if keyEvent.uChar == 0:
+        # ignore the first SHIFT_PRESSED event - the subsequent read
+        # has the result. Details:
+        # https://github.com/nim-lang/Nim/issues/7764
+        if (keyEvent.dwControlKeyState and SHIFT_PRESSED) == SHIFT_PRESSED:
+          continue
+      else: discard        
```

Comment:
btw this ``else: discard`` is entirely redundant

Reply:
Thanks for that - forgot to remove the leftover `else` from the previous `case` statement.